File: Javascript/DSA/queue.js
function Queue() {
    var items = [];

    this.enqueue = function(element){
        items.push(element);
    }

    this.dequeue = function(element) {
        items.shift();
    }

    this.front = function() {
        return items[0];
    }

    this.isEmpty = function() {
        return items.length == 0;
    }

    this.clear = function() {
        items = [];
    }

    this.size = function() {
        return items.length;
    }

    this.print = function() {
        console.log(items.toString());
    }
}

let queue = new Queue();

queue.enqueue("Cat");
queue.enqueue("Dog");
queue.enqueue("Shark");
queue.print();
queue.dequeue();
queue.print();
console.log(queue.size());
console.log(queue.isEmpty());
queue.print();
queue.dequeue();
queue.print();
console.log(queue.size());
console.log(queue.isEmpty());
queue.clear();
queue.print();
console.log(queue.size());

File: C#/DSA/Queue.cs
using DSA.util;
using System.Collections;

namespace DSA
{
    public class Queue<T>
    {

        private readonly ArrayList array;
        private int count;

        public Queue()
        {
            array = new ArrayList();
            count = 0;
        }

        public void Enqueue(T element)
        {
            array.Add(element);
            count++;
        }
        public void Dequeue()
        {
            array.RemoveAt(0);
            count--;
        }

        public int Size()
        {
            return count;
        }

        public bool IsEmpty()
        {
            return count == 0;
        }

        public T Front()
        {
            return (T)array[0];
        }

        public override string ToString()
        {
            return JSon.ConvertToJson(array);
        }
    }
}
